app/scripts/metamask-controller.test.js

```diff
@@ -1287,6 +1287,127 @@ describe('MetaMaskController', () => {
           );
         });
       });
+
+      it.todo('should only process `metamask-provider` multiplex formatted messages');
```

Comment:
Not really able to test this for some reason. I can see outside messages from the spec hitting the `createThroughStream` stream, but it never reaches the logger. Additionally, I see output messages written to this stream also getting picked back up as if it was an incoming message. These mocked streams need to be updated, and the existing specs fixed before we can probably test this

Reply:
What's the solution here? Are you recommending that we do update the mock streams in a subsequent PR?

Reply:
exactly. we need a follow up ticket since there is unrelated pre-req work that must be completed first. I mean, we could do it as part of this PR, but i think it's best that we try to keep commits into dev as focused as possible